test/functional/neutronless/bigip_interaction.py

```diff
@@ -197,26 +188,74 @@ def check_resulting_cfg(cls, test_name=current_test):
 
         test_name := the name of the test currently in tearDown
         """
-        cls.check_symbols()
-        cls._resulting_bigip_cfg(test_name)
+        if test_name:
+            cls.__current_test = test_name
+        else:
+            test_name = cls.__current_test
+        if not hasattr(pytest.symbols, 'no_bigip_tracking') and cls.ssh_cmd:
+            cls._resulting_bigip_cfg(test_name)
+        elif not hasattr(pytest.symbols, 'no_bigip_tracking') and \
+                not cls.ssh_cmd:
+            raise EnvironmentError("Broken state... cannot track without "
+                                   "ssh_cmd!")
 
     @classmethod
-    def backup_bigip_cfg(cls):
+    def backup_bigip_cfg(cls, test_name):
         """Performs a config backup of the BIG-IP's configuration
 
         This method will store a backup of the BIG-IP's configuration on the
         BIG-IP for later restoration.
         """
-        if not os.path.isfile(cls.config_file.format(my_epoch)):
-            cls.check_symbols()
+        cls.__current_test = test_name
+        if not os.path.isfile(cls.config_file.format(my_epoch)) and \
+                not hasattr(pytest.symbols, 'no_bigip_tracking') and \
+                cls.ssh_cmd:
             cls.__exec_shell(
                 cls.__ucs_cmd_fmt.format(cls.ssh_cmd, 'save'), True)
             cls._get_existing_bigip_cfg()
+        elif not hasattr(pytest.symbols, 'no_bigip_tracking') and \
+                not cls.ssh_cmd:
+            raise EnvironmentError("Broken state... cannot track without "
+                                   "ssh_cmd!")
+
 
+def BEGIN():
+    """Performs library's initial, imported setup
 
-@pytest.fixture
-def track_bigip_cfg(request):
-    global current_test
-    request.addfinalizer(BigIpInteraction.check_resulting_cfg)
-    BigIpInteraction.backup_bigip_cfg()
-    current_test = request.node.name  # current test's scoped name
+    This setup function will perform basic consolidation in meaning provided in
+    environment variables.
+
+    User Notes: if it is found that bigip config tracking is unnecessary and is
+    reducing or impacting production time, an added 'no_bigip_tracker' variable
+    can be added to the --symbols <file>.  In this instance, this library will
+    not execute, and all fixtures that use this library will not perform
+    bigip config tracking.
+
+    WARNING: Nightly will perform this tracking; thus, a user does no justice
+    if linging config is later discovered.  It is; therefore, still recommended
+    to run this as a last step to assure proper config tracking is assured.
+    """
+    ssh_options = """ssh -o StrictHostKeyChecking=no \
+    -o UserKnownHostsFile=/dev/null \
+    {}"""
+    ssh_host_specific_fmt = "{}@{}"
+    ssh_cmd = ''
+    hostname = ''
+    username = ''
+    if 'TESTENV_AGENT_BRANCH' in os.environ:
```

Comment:
I think we should use symbols as the sole input interface for the openstack tests.  That is/was the primary purpose of the pytest-symbols plugin.

The only time testenv should come up in conversation is when we talk about whether or not it was used to produce the symbols file.

The tests themselves shouldn't have to know about (or mention) testenv.